File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} threshold
 * @return {number}
 */
var numOfSubarrays = function(arr, k, threshold) {
    let window_sum = 0;
    let count = 0 , n = arr.length;
    for(let i = 0; i < k ; i++)
        {
            window_sum+=arr[i];
        }
    for(let i = 0 ; i < n+k-1 ; i++)
        {
            if(i != 0)
                {
                    window_sum -= arr[i - 1];
                    window_sum += arr[i + k -1];
                }
            let window_avg = window_sum / k;
            if(window_avg >= threshold)
                count++;
        }
    return count;
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) 
    {
        int window_sum = 0;
        int count = 0;
        int n = arr.size(),res = 0;
        for(int i = 0 ; i< k;i++)
            window_sum += arr[i];//Calculating the sum of the frist window
        for(int i = 0 ; i< n - k + 1 ; i++)
        {
            if(i != 0)
            {
                window_sum -= arr[i-1];
                window_sum += arr[i + k -1];
                //new window at each iteration is ready
            }
            int window_avg = window_sum / k;
            if(window_avg >= threshold)
                res += 1;
        }
        return res;
    }
};